### PR TITLE
change default aws linux instance type to t3.micro, reduce default template TTL

### PR DESCRIPTION
**Pull request by johnstcn**

### CODE REVIEW

File: cli/templatecreate.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func templateCreate() *cobra.Command {
 	cmd.Flags().StringVarP(&directory, "directory", "d", currentDirectory, "Specify the directory to create from")
 	cmd.Flags().StringVarP(&provisioner, "test.provisioner", "", "terraform", "Customize the provisioner backend")
 	cmd.Flags().StringVarP(&parameterFile, "parameter-file", "", "", "Specify a file path with parameter values.")
-	cmd.Flags().DurationVarP(&maxTTL, "max-ttl", "", 168*time.Hour, "Specify a maximum TTL for worksapces created from this template.")
+	cmd.Flags().DurationVarP(&maxTTL, "max-ttl", "", 8*time.Hour, "Specify a maximum TTL for workspaces created from this template.")
```
Comment (johnstcn) [inline cli/templatecreate.go:130]:
```suggestion
	cmd.Flags().DurationVarP(&maxTTL, "max-ttl", "", 2*time.Hour, "Specify a maximum TTL for workspaces created from this template.")
```
```suggestion
	cmd.Flags().DurationVarP(&maxTTL, "max-ttl", "", 8*time.Hour, "Specify a maximum TTL for workspaces created from this template.")
```

File: cli/templatecreate.go [hunk 2]
```diff
@@ -127,7 +127,7 @@ func templateCreate() *cobra.Command {
 	cmd.Flags().StringVarP(&directory, "directory", "d", currentDirectory, "Specify the directory to create from")
 	cmd.Flags().StringVarP(&provisioner, "test.provisioner", "", "terraform", "Customize the provisioner backend")
 	cmd.Flags().StringVarP(&parameterFile, "parameter-file", "", "", "Specify a file path with parameter values.")
-	cmd.Flags().DurationVarP(&maxTTL, "max-ttl", "", 168*time.Hour, "Specify a maximum TTL for worksapces created from this template.")
+	cmd.Flags().DurationVarP(&maxTTL, "max-ttl", "", 2*time.Hour, "Specify a maximum TTL for workspaces created from this template.")
```
Comment (kylecarbs) [inline cli/templatecreate.go:130]: Do we display the max TTL in the UI anywhere? I'm concerned about setting this so low without a clear indicator to the user.

Comment (johnstcn) [reply cli/templatecreate.go:130]: That's fair!

File: examples/templates/aws-linux/main.tf [hunk 3]
```diff
@@ -130,7 +130,7 @@ EOT
 resource "aws_instance" "dev" {
   ami               = data.aws_ami.ubuntu.id
   availability_zone = "${var.region}a"
-  instance_type     = "t3.xlarge"
+  instance_type     = "t3.micro"
```
Comment (spikecurtis) [inline examples/templates/aws-linux/main.tf:133]: This is going to give people a poor experience with cloud workspaces.  I understand the desire to keep our own spend low when these things get orphaned, but I think we need to do this in a way that doesn't cripple end users experience.

We want people to have a good experience with cloud development and get hooked.  A micro is less powerful than my phone.

**Review by spikecurtis — CHANGES_REQUESTED** [review]

File: examples/templates/aws-linux/main.tf (hunk 3 above)
Comment (johnstcn) [reply examples/templates/aws-linux/main.tf:133]: This bring it in line with the AWS windows template. Should be instead update the AWS windows template to match?
Alternatively, we could instead force the user to choose an instance type when creating a workspace via the template.
What do you think?

Comment (spikecurtis) [reply examples/templates/aws-linux/main.tf:133]: I think modeling it as a variable is a good idea for our example templates.  Give them a reasonable range but keep the default something powerful.  Hopefully our own devs who are just testing, not dogfooding, will choose something small.

File: examples/templates/aws-linux/main.tf [hunk 4]
```diff
@@ -36,6 +36,22 @@ variable "region" {
   }
 }
 
+variable "instance_type" {
+  description = "What instance type should your workspace use?"
+  default     = "2C/1G: t3.micro"
```
Comment (mafredri) [inline examples/templates/aws-linux/main.tf:41]: I've had a few instances where defining the `default` skipped the prompt. I'm guessing this is not the case here? (Just checking.)

**Review by mafredri — APPROVED** [review]
Default TTL of 2h sounds a bit short, but I understand why we're making the change, so all good 👍🏻.

File: examples/templates/aws-linux/main.tf (hunk 4 above)
Comment (johnstcn) [reply examples/templates/aws-linux/main.tf:41]: I don't recall it being the case when I tested it!

File: examples/templates/aws-linux/main.tf [hunk 5]
```diff
@@ -36,6 +36,22 @@ variable "region" {
...
+  validation {
+    condition = contains([
+      "2C/1G: t3.micro",
+      "2C/2G: t3.small",
```
Comment (spikecurtis) [inline examples/templates/aws-linux/main.tf:45]: It wasn't immediately clear what "C" and "G" meant in this context.  I eventually realized it was vCPU and GB RAM, but I think 2 vCPU /  2 GB RAM is clearer and it's not like we're hurting for space in this context, nor do the users have to type these values in our UX.

Unfortunately the vCPU number is a little misleading for T3 instances. Micro-large have the same number of vCPUs, and reading this you might conclude that they have the same CPU performance.  They do not.  Amazon throttles the CPUs except for short bursts and the amount of throttling is different for the different instance sizes.  I'm not really sure what to suggest except to drop the stats and just include the instance type names.  Up to you.

**Review by spikecurtis — APPROVED** [review]

File: examples/templates/aws-linux/main.tf (hunk 5 above)
Comment (johnstcn) [reply examples/templates/aws-linux/main.tf:45]: I can drop the stats, I only put them in there so the instance types would show up in the right order :D

Comment (mafredri) [reply examples/templates/aws-linux/main.tf:45]: I guess you could do numbered list, `1. t3.micro`, `2. t3.small` if you want to maintain order? 😄

**Pull request merged**